### ISSUE TITLE
[Feature Request] Confirmation dialog when hitting Esc on Interactive Rebase

### ISSUE THREAD

**Issue by fgimian** [ForkIssues/TrackerWin#255]
It's currently way too easy to lose your work by accidentally hitting Esc after making many edits on an interactive rebase.  This just happened to me and honestly I don't even know how, but I lost a good 15 mins of work.

Kindly consider a confirmation dialog if any changes have been made and a user tries to cancel an interactive rebase.

Thanks heaps and great job on this app!!! 😄 
Fotis

**Comment by fgimian** [comment]
Actually, it turns out I hadn't hit Esc as I reproduced the issue again.  After renaming a bunch of commits and scrolling through to check my work, the Interactive Rebase dialog box just disappeared and I lost all my work again 😢 

I'm going back to the CLI for now 😄

**Comment by DanPristupov** [comment]
Most probably you hit 5 minute timeout for an active rebase process. I need to improve that.

**Comment by fgimian** [comment]
> Most probably you hit 5 minute timeout for an active rebase process. I need to improve that.

Ooooh, that makes sense.  Yes, please remove the timeout completely if you can.  On a large rebase, it can easily take longer than that.

Cheers
Fotis

**Comment by vadsiraly** [comment]
Hah! We are not alone. I second this.

**Comment by DanPristupov** [comment]
Yes, we have this already: https://github.com/ForkIssues/TrackerWin/issues/186.

I will think how to fix that.

**Comment by fgimian** [comment]
Thanks @vadsiraly, looks like you beat me to it with your report 😄 

Personally, I think there should *never* be a timeout on such operations.  When the user is finished, they will close the dialog box and the process can die.

I hope there's a way to implement this and remove the timout completely, this is extremely strange behaviour imho.

Huge love and thanks
Fotis

**Comment by DanPristupov** [comment]
> Personally, I think there should never be a timeout on such operations. 

This is dangerous. IR is performed as a background process. So, if you kill Fork in Task Manager (or it crashes), the repository will be blocked until a reboot.

> I hope there's a way to implement this and remove the timeout completely, this is extremely strange behaviour imho.

Any timeout could be considered as a strange behaviour, because usually it handles cases when something went wrong.

**Comment by fgimian** [comment]
Really appreciate the reply and your commitment to helping solve this!

Ultimately as a user, I simply can't trust Fork for interactive rebases in its current form because I've lost my work twice when the timeout was enforced.  The dialog just diassppeared even though I had renamed dozens of commits.

It's way too risky, especially for such tedious operations that I would never want to do more than once (in the case above, I did it 3 times, twice with Fork which timed out both times ... and then finally I just used the CLI and everything was fine).  Lucikly I had put my old and new commit messages into a file in Sublime text and was copy / pasting, but even so, it was still very frustrating doing that multiple times.

Hope that a solution can be found to remedy this because Fork otherwise does a wonderful job on interactive rebase 😄
